src/components/select/select-list/select-list.component.js

```diff
@@ -93,63 +107,92 @@ const SelectList = React.forwardRef(
       [anchorElement]
     );
 
-    const optionRefList = useMemo(
-      () =>
-        React.Children.map(children, (child) => {
-          if (child?.type === Option || child?.type === OptionRow) {
-            return React.createRef();
-          }
-          return null;
-        }).filter((child) => child),
-      [children]
-    );
-
     const handleSelect = useCallback(
       (optionData) => {
         onSelect({ ...optionData, selectionType: "click" });
       },
       [onSelect]
     );
 
-    const childIds = useMemo(() => React.Children.map(children, () => guid()), [
+    const childIdsRef = useRef(null);
+
+    // childIds should be stable except when children are added or removed - can't use useMemo
+    // as that isn't absolutely guaranteed to never rerun when dependencies haven't changed.
+    const setChildIds = () => {
```

Comment:
yes, I realised it did because we don't want to replace the IDs except when the children change - useMemo doesn't guarantee this (even though it likely works in practice), it's supposed to be a performance optimisation, not a guarantee of memoisation for functionality purposes which is what we were using it for here.